.github/workflows/publish.yaml

```diff
@@ -0,0 +1,55 @@
+name: Publish ROCK
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup Docker
+        run: |
+          sudo snap install docker
+          sudo addgroup --system docker; sudo adduser $USER docker
+          newgrp docker
+          sudo snap disable docker; sudo snap enable docker
```

Comment:
Is this doing something?

Reply:
Installing docker for tagging and pushing the image to GHCR. Do you mean a specific line here or in general?

Reply:
Line 24, where it disables and enables `docker`.

Reply:
With the snap you need to disable and enable docker after adding the new group if it was added while the snap was enabled: https://askubuntu.com/questions/941816/permission-denied-when-running-docker-after-installing-it-as-a-snap